haystack/components/generators/hf_utils.py

```diff
@@ -61,6 +63,43 @@ def check_valid_model(model_id: str, token: Optional[Secret]) -> None:
         raise ValueError(f"Model {model_id} is not a text generation model. Please provide a text generation model.")
 
 
+def inference_deployed_models(headers: Optional[Dict] = None) -> List[Dict[str, Any]]:
+    """
+    Get all currently deployed models on HF TGI free tier
+
+    :param headers: Optional dictionary of headers to include in the request
+    :type headers: Optional[Dict]
+    :return: list of all currently deployed models
+    :raises Exception: If the request to the TGI API fails
+
+    """
+    resp = requests.get(
+        "https://api-inference.huggingface.co/framework/text-generation-inference", headers=headers, timeout=5
+    )
+
+    payload = resp.json()
+    if resp.status_code != 200:
+        message = payload["error"] if "error" in payload else "Unknown TGI error"
+        error_type = payload["error_type"] if "error_type" in payload else "Unknown TGI error type"
+        raise Exception(f"Failed to fetch TGI deployed models: {message}. Error type: {error_type}")
+
+    return payload
+
+
+def list_inference_deployed_models(headers: Optional[Dict] = None) -> List[str]:
+    """
+    List all currently deployed models on HF TGI free tier
+
+    :param headers: Optional dictionary of headers to include in the request
+    :type headers: Optional[Dict]
+    :return: list of all currently deployed models
+    :raises Exception: If the request to the TGI API fails
+
+    """
+    deployed_models = inference_deployed_models(headers)
+    return [model["model_id"] for model in deployed_models]
```

Comment:
I would merge these functions. `inference_deployed_models` is never used outside of `list_inference_deployed_models`.

I would make `inference_deployed_models` return the list of model ids directly. As of now it only creates indirection.

Reply:
Ok will do, thank you @silvanocerza